File: generated-code/example/fsharp/Runner.fs
namespace TransGenTest

open System
open System.IO

module Runner =

    [<EntryPoint>]
    let main argv =
        if argv.Length <> 2 then
            failwith "Pass input and output as parameters"
        let inputFile = argv.[0]
        let outputFile = argv.[1]

        use inputStream = new FileStream(inputFile, FileMode.Open)
        use reader = new BinaryReader(inputStream)
        let input = Model.Structure.readFrom(reader)

        use outputStream = new FileStream(outputFile, FileMode.Create)
        use writer = new BinaryWriter(outputStream)
        input.writeTo(writer)

        0

File: generated-code/example/csharp/Runner.cs
using System;
using System.IO;

namespace TransGenTest
{
    public class Runner
    {
        public static void Main(string[] args)
        {
            if (args.Length != 2)
            {
                throw new Exception("Pass input and output as parameters");
            }
            string inputFile = args[0];
            string outputFile = args[1]; 

            Model.Example input;
            using (var stream = new FileStream(inputFile, FileMode.Open))
            {
                using (var reader = new BinaryReader(stream))
                {
                    input = Model.Example.ReadFrom(reader);
                }
            }
            Console.WriteLine(input.ToString());
            using (var stream = new FileStream(outputFile, FileMode.Create))
            {
                using (var writer = new BinaryWriter(stream)) {
                    input.WriteTo(writer);
                }
            }
        }
    }
}